traitsui/testing/tests/test_ui_tester.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+import unittest
+
+from traits.api import (
+    Bool, Button, Date, Enum, Instance, HasTraits, List, Str, Int,
+    on_trait_change, Property
+)
+from traitsui.api import EnumEditor, Item, ModelView, View
+from traitsui.testing.api import Disabled, UITester
+from traitsui.tests._tools import (
+    requires_one_of,
+    QT,
+    WX,
+)
+
+
+class Parent(HasTraits):
+
+    first_name = Str()
+    last_name = Str()
+    age = Int()
+    employment_status = Enum(["employed", "unemployed"])
+
+    def default_traits_view(self):
+        return View(
+            Item("first_name"),
+            Item("last_name"),
+            Item("age"),
+            Item(
+                "employment_status",
+                editor=EnumEditor(
+                    values=["employed", "unemployed"],
+                    evaluate=True,
+                    auto_set=False,
+                )
+            ),
+        )
+
+
+class Child(HasTraits):
+
+    mother = Instance(Parent)
+    father = Instance(Parent)
+    first_name = Str()
+    last_name = Property(depends_on="father,father.last_name")
+
+    def _get_last_name(self):
+        if self.father is None:
+            return None
+        return self.father.last_name
+
+
+class SimpleApplication(ModelView):
+
+    model = Instance(Child)
+
+
+@requires_one_of([QT, WX])
+class TestUITesterIntegration(unittest.TestCase):
+    """ Integration tests for UITester.
+
+    These tests are close imitations for how UITester is expected to be used.
+    """
+
+    def setUp(self):
+        # UITest can also be used as a context manager
+        self.tester = UITester()
+        self.tester.start()
+
+    def tearDown(self):
+        self.tester.stop()
+
+    def test_instance_text_editor_from_view(self):
+        # Test popagating commands to instance simple/custom editors
+        father = Parent(first_name="M", last_name="C")
+        mother = Parent(first_name="J", last_name="E", age=50)
+        child = Child(mother=mother, father=father)
+        view = View(
+            Item("model.father", style="simple"),
+            Item("model.mother", style="custom"),
+        )
+        app = SimpleApplication(model=child)
+        with self.tester.create_ui(app, dict(view=view)) as ui:
+            self.tester.set_text(ui, "father.first_name", "B")
```

Comment:
I see your point there.
It is possible for the tester to keep a reference to the UI to remove that one argument, but we should probably then add an optional argument to override the default behaviour in case one really needs more than one UI at the same time, because saving one positional argument does not seem worth it to kill that possibility altogether.
On the other hand, the code is more explicit when `ui` is passed. The code literally reads "set the text of ui's 'father.first_name' to 'B'".  Explicit is better than implicit so I am still inclined towards keeping `ui` here. I can be convinced though.

Reply:
I don't have strong feelings about this. Mainly I was struggling to fit `as ui` into one line so I thought maybe we can get rid of it 😄

Reply:
Let's file a petition to increase the maximum line length for flake8 :D